packages/x-components/src/x-modules/facets-next/entities/__tests__/editable-number-range-filter-entity.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { createLocalVue } from '@vue/test-utils';
+import Vuex, { Store } from 'vuex';
+import { EditableNumberRangeFilter } from '@empathyco/x-types';
+import { createNextEditableNumberRangeFilter } from '../../../../__stubs__/filters-stubs.factory';
+import { RootXStoreState } from '../../../../store/store.types';
+import { facetsNextXStoreModule } from '../../store/module';
+import { EditableNumberRangeFilterEntity } from '../editable-number-range-filter.entity';
+
+describe('testing EditableNumberRangeFilterEntity', () => {
+  function prepareFacetsStore(): Store<RootXStoreState> {
+    const vue = createLocalVue();
+    vue.use(Vuex);
+    return new Store({
+      modules: {
+        x: {
+          modules: {
+            facetsNext: { ...facetsNextXStoreModule, namespaced: true }
+          },
+          namespaced: true
+        }
+      }
+    });
+  }
+
+  it('allows selecting a filter when filter min is not null', () => {
+    const store = prepareFacetsStore();
+    const filter = createNextEditableNumberRangeFilter('price', { min: 3, max: null });
+    const filterEntity = new EditableNumberRangeFilterEntity(store, filter);
+
+    filterEntity.select();
+    expect(store.state.x.facetsNext.filters[filter.id].selected).toBe(true);
+  });
+
+  it('allows selecting a filter when filter max is not null', () => {
+    const store = prepareFacetsStore();
+    const filter = createNextEditableNumberRangeFilter('price', { min: null, max: 5 });
+    const filterEntity = new EditableNumberRangeFilterEntity(store, filter);
+
+    filterEntity.select();
+    expect(store.state.x.facetsNext.filters[filter.id].selected).toBe(true);
+  });
+
+  it('allows deselecting a filter and set its range values to null', () => {
+    const store = prepareFacetsStore();
+    const filter = createNextEditableNumberRangeFilter('price', { min: 2, max: 10 });
+    const filterEntity = new EditableNumberRangeFilterEntity(store, filter);
+
+    filterEntity.select();
+    expect(store.state.x.facetsNext.filters[filter.id].selected).toBe(true);
+
+    filterEntity.deselect();
+    expect(store.state.x.facetsNext.filters[filter.id].selected).toBe(false);
+    expect(
```

Comment:
Because of the verbosity of these tests, we created a `isFilterSelected` method, I'm going to move it to the `utils.ts` right now. Maybe you could do the same for the range? Just to avoid casting too many times, like @tajespasarela is suggesting for the entity code.

Reply:
instead of only using the `isFilterSelected` function, maybe it is better to create a new `isRangeSelected` function where to check, not only if the `selected` field is `true` or `false`, but also if the `range` has or not values.